### PR TITLE
checker: Add a tool to check MySQL schema compatibility with TiDB

### PR DESCRIPTION
**Pull request by shenli**
@qiuyesuifeng @siddontang @iamxy PTAL

### CODE REVIEW

File: checker/main.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"flag"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+)
+
+var (
+	logLevel = flag.String("L", "info", "log level: info, debug, warn, error, fatal")
```
Comment (siddontang) [inline checker/main.go]: if we use error level, we will ignore foreign key warning.

File: checker/main.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func main() {
+	flag.Parse()
+	if len(flag.Args()) == 0 {
+		log.Error("Miss database name")
+		return
+	}
+	log.SetLevelByString(*logLevel)
+	c := &checker{
+		dbName: flag.Args()[0],
```
Comment (siddontang) [inline checker/main.go]: if we pass `exec --host 127.0.0.1 db tbls...`, the first argument is db?

Comment (shenli) [inline checker/main.go]: Yes, just follow mysqldump style.

File: checker/main.go (hunk 1 above)
Comment (shenli) [inline checker/main.go]: Will fix.

**Comment by shenli** [comment]
1. Address comment.
2. Fix typo in README.md.
3. Check rest tables when meet error.

@siddontang @tiancaiamao PTAL

**Comment by siddontang** [comment]
LGTM

File: checker/main.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		tbls:   flag.Args()[1:],
+	}
+	err := c.check()
+	if err != nil {
+		log.Errorf("Check database error: %s", err)
+	} else if c.warnings > 0 || c.errs > 0 {
+		log.Errorf("Check database %s with %d errors and %d warnings.", c.dbName, c.errs, c.warnings)
```
Comment (iamxy) [inline checker/main.go]: add a `os.Exit(2)` following is better ?
the checker may be used in a shell script.

File: checker/checker.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	return nil
+}
+
+func (c *checker) getCreateTable(tn string) (string, error) {
+	rs, err := querySQL(c.db, fmt.Sprintf("show create table %s;", tn))
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	defer rs.Close()
+	for rs.Next() {
```
Comment (tiancaiamao) [inline checker/checker.go]: if it's just one line, why not use `if` instead of `for`

File: checker/main.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+var (
+	logLevel = flag.String("L", "info", "log level: info, debug, warn, error, fatal")
+	host     = flag.String("host", "127.0.0.1", "MySQL host")
+	port     = flag.Int("port", 3306, "MySQL port")
+	username = flag.String("user", "root", "User name")
+	password = flag.String("password", "", "Password")
+)
+
+func createDB(dbName string) (*sql.DB, error) {
```
Comment (tiancaiamao) [inline checker/main.go]: createDB is not a good name for this function, openDB or connectToDB would be better.

**Comment by shenli** [comment]
@iamxy @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**